cpp/src/buffer/resource.cpp

```diff
@@ -155,18 +156,12 @@ std::unique_ptr<Buffer> BufferResource::allocate(
     return allocate(reservation.size(), stream, reservation);
 }
 
-std::unique_ptr<Buffer> BufferResource::move(
-    std::unique_ptr<std::vector<uint8_t>> data, rmm::cuda_stream_view stream
-) {
-    return std::unique_ptr<Buffer>(new Buffer(std::move(data), stream));
-}
-
 std::unique_ptr<Buffer> BufferResource::move(
     std::unique_ptr<rmm::device_buffer> data, rmm::cuda_stream_view stream
 ) {
     auto upstream = data->stream();
     if (upstream.value() != stream.value()) {
-        cuda_stream_join(std::array{stream}, std::array{upstream});
+        cuda_stream_join(std::views::single(stream), std::views::single(upstream));
```

Comment:
*question*: this is so common, should we provide an overload for `cuda_stream_join`?